src/database.ts

```diff
@@ -1,88 +1,77 @@
 /* eslint no-console: 0 */
-import * as mysql from 'mysql2/promise';
-import { readdir } from 'fs/promises';
-import { join } from 'path';
+import { knex, Knex } from 'knex';
 
-let pool: mysql.Pool;
+let pool: Knex;
 
-async function getPool() {
+async function getPool(): Promise<Knex> {
 
   if (!pool) {
-    pool = mysql.createPool(await getSettings());
+    pool = knex(await getSettings());
   }
 
   return pool;
-
-}
-
-
-export async function getConnection(): Promise<mysql.PoolConnection> {
-
-  return (await getPool()).getConnection();
-
 }
 
-export async function query(query: string, params?: any[]|string|Record<string,any>): Promise<[any, any]> {
+export async function getConnection(): Promise<Knex> {
 
-  return (await getPool()).query(query, params);
+  return (await getPool());
 
 }
 
-export async function checkPatches() {
-
-  // Get a list of known database patches.
-  const files = await readdir(join(__dirname, '..', 'mysql-schema'));
-  const patches = new Map<number, string>();
-  for(const file of files) {
-    const match = file.match(/^([0-9]+)-.*\.sql$/);
-    if (!match) continue;
-    patches.set(
-      parseInt(match[1]),
-      file
-    );
-  }
-
-  const result = await query('SELECT id FROM changelog');
-  for (const { id } of result[0]) {
-    if (!patches.has(id)) {
-      console.warn('Warning! Found an unknown database patch in the changelog. Patch id: ' + id);
-    }
-    patches.delete(id);
-  }
+export async function query<T = any>(query: string, params: Knex.ValueDict | Knex.RawBinding[] = []): Promise<Knex.Raw<T>> {
 
-  if (patches.size > 0) {
-    console.error('Error!! Database patches have not been applied. Please run the following .sql scripts:');
-    for (const fileName of patches.values()) {
-      console.error(`  - ${fileName}`);
-    }
-    throw new Error('Missing patches');
-  }
+  return (await getPool()).raw(query, params);
 
 }
 
 export default {
-  query,
   getConnection,
-  checkPatches,
+  query,
 };
 
 async function getSettings() {
 
-  let settings: any = {};
-
-  // We are running in a local environment.
-  settings = {
-    user: process.env.MYSQL_USER,
-    password: process.env.MYSQL_PASSWORD,
-    database: process.env.MYSQL_DATABASE,
-  };
+  let connection: any = {};
+  let client: string = '';
```

Comment:
```suggestion
  let connection;
  let client;
```

I assume if you don't initialize these, typescript will complain. That's a good thing, because I think you just need an `else` statement with an explicit error.

Reply:
This is not 100% resolved, because initializing these variables is still a bad idea.

Reply:
Ohhh I didnt fully understand your initial comment before! I'll fix